### PR TITLE
feat: ✨ support scroller resize direction

### PR DESCRIPTION
**Pull request by MrMengJ**
Closes: #2323

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Enhancement (changes that improvement of current feature or performance)
- [ ] Refactoring (changes that neither fixes a bug nor adds a feature)
- [ ] Test Case (changes that add missing tests or correct existing tests)
- [ ] Code style optimization (changes that do not affect the meaning of the code)
- [ ] Docs (changes that only update documentation)
- [ ] Chore (changes that don't modify src or test files)

### Self Check before Merge

- [ ] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING**](https://github.com/antvis/x6/blob/master/CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/x6/src/addon/scroller/index.ts [hunk 1]
```diff
@@ -396,12 +396,73 @@ export class Scroller extends View {
...
       allowNewOrigin: 'negative',
+      contentArea: this.calcContextArea(),
       ...resizeOptions,
     }
 
     this.graph.fitToContent(this.getFitToContentOptions(options))
   }
 
+  protected calcContextArea() {
+    let resizeOptions =
```
Comment (NewByVector) [inline packages/x6/src/addon/scroller/index.ts:407]: 这里的 resizeOptions 在 390 行已经计算过一次，直接作为 calcContextArea 的参数传过来就行。

**Review by NewByVector — CHANGES_REQUESTED** [review]

**Review by NewByVector — CHANGES_REQUESTED** [review]
409 行的 resizeOptions 在 390 行已经计算过一次，直接作为 calcContextArea 的参数传过来就行。

**Review by NewByVector — APPROVED** [review]

File: packages/x6/src/addon/scroller/index.ts [hunk 2]
```diff
@@ -1201,17 +1265,25 @@ export namespace Scroller {
      * @deprecated
      */
     fitTocontentOptions?:
-      | TransformManager.FitToContentFullOptions
+      | (TransformManager.FitToContentFullOptions & {
+          direction: AutoResizeDirection | AutoResizeDirection[]
```
Comment (OpportunityLiu) [inline packages/x6/src/addon/scroller/index.ts:1269]: 这里没有设为可选，导致以前的代码编译不过了。。

File: packages/x6/src/addon/scroller/index.ts [hunk 3]
```diff
@@ -396,12 +396,76 @@ export class Scroller extends View {
...
+          direction:
+            | Scroller.AutoResizeDirection
+            | Scroller.AutoResizeDirection[]
+        })
+      | undefined,
+  ) {
+    const direction = resizeOptions?.direction
+
+    if (!direction) {
+      return this.graph.transform.getContentArea()
```
Comment (OpportunityLiu) [inline packages/x6/src/addon/scroller/index.ts:418]: 今天又发现一个问题，这里没有把 `resizeOptions` 传进去，导致我在 `autoResizeOptions` 给的 `useCellGeometry: true` 不起作用

File: packages/x6/src/addon/scroller/index.ts [hunk 4]
```diff
@@ -396,12 +396,76 @@ export class Scroller extends View {
...
+    }
+
+    function getCellBBox(cell: Cell) {
```
Comment (OpportunityLiu) [inline packages/x6/src/addon/scroller/index.ts:421]: 这些实现好像又默认 `useCellGeometry: true` 了，没使用 `Dom.getBBox` 之类的 API

**Pull request merged**